src/java.base/share/classes/sun/nio/cs/StreamDecoder.java

```diff
@@ -262,6 +262,10 @@ public void fillZeroToPosition() throws IOException {
         bb.flip();
     }
 
+    public String readString() throws IOException {
+        return new String(in.readAllBytes(), decoder.charset());
```

Comment:
I think we need to handle leftover chars here, including the remaining char and the remaining unprocessed bytes (in case of underflow) in the CharsetDecoder